utils/codepointtrie/src/codepointtrie.rs

```diff
@@ -797,6 +775,83 @@ impl<'trie, T: TrieValue + Into<u32>> CodePointTrie<'trie, T> {
             cpm_range: init_range,
         }
     }
+
+    /// Yields an [`Iterator`] returning the ranges of the code points whose values
+    /// match `value` in the [`CodePointTrie`].
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use icu_codepointtrie::planes;
+    ///
+    /// let trie = planes::get_planes_trie();
+    ///
+    /// let plane_val = 2;
+    /// let mut sip_range_iter = trie.get_ranges_for_value(plane_val as u8);
+    ///
+    /// let start = plane_val * 0x1_0000;
+    /// let end = start + 0xffff;
+    ///
+    /// let sip_range = sip_range_iter.next()
+    ///     .expect("Plane 2 (SIP) should exist in planes data");
+    /// assert_eq!(start..=end, sip_range);
+    ///
+    /// assert!(sip_range_iter.next().is_none());
+    pub fn get_ranges_for_value(&self, value: T) -> impl Iterator<Item = RangeInclusive<u32>> + '_ {
```

Comment:
It might be easier to implement `gc=Letter` if this takes a closure (to use in the `.filter()` call) instead of a value. The simple cases would use `|range| range.value == value` like here, but you could also, for example, check to see if `range.value` is included in a GeneralCategoryGroup.

I'm okay landing this and adding the extended version in a separate PR.